test/Elastic.Apm.Tests.MockApmServer/AssertValidExtensions.cs

```diff
@@ -217,7 +217,16 @@ internal static void AssertValid(this CapturedException thisObj)
 
 			thisObj.Code?.AssertValid();
 			thisObj.StackTrace?.AssertValid();
-			thisObj.Type.AssertValid();
+
+			if (string.IsNullOrEmpty(thisObj.Type))
```

Comment:
Is there a reason for special treatment for empty string? it seems intake API spec treats it as valid.

Reply:
This is related to this one: https://github.com/elastic/apm-agent-dotnet/pull/450/files#r317177339 

I'm trying here to make sure either `Type` or `Message` is set - either one of those or both. Both can't be empty.